packages/ia-zendesk-help-widget/src/styles/zendesk-widget.js

```diff
@@ -0,0 +1,43 @@
+import { css } from 'lit-element';
+
+export default css`
+  .help-widget {
+    padding: 13px 22px;
+    margin: 14px 20px;
+    position: fixed;
+    bottom: 0px;
+    z-index: 999998;
+    transition-duration: 250ms;
+    right: 0px;
+    background: #194880;
+    color: #FFFFFF;
+    letter-spacing: 0.6px;
+    font-size: inherit;
+    transition: opacity 0.12s linear;
+    border-radius: 999rem;
+    border: 0;
+    outline: none;
+    font-weight: 700;
+    cursor: pointer;
+    vertical-align: middle;
+  }
+
+  .help-widget svg {
+    vertical-align: middle;
+    margin-right: 3px;
+  }
+
+  @media (max-width: 767px) {
```

Comment:
In general, I would try to avoid setting state inside a component that is dependent on something happening outside of the component, in this case, the window being resized.

Since it looks like you're hiding the "Help" text when the window gets smaller, I would treat this as 2 different modes, minimal and full, and make that a property on the webcomponent itself, ie:
```
buttonMode: { type: String } // with values of "minimal" or "full"
```
then set a class on your component based on the mode, ie:
```
<div class="${this.buttonMode}">...</div>
```
set up your CSS:
```
css`
  .minimal .help-text {
    .display: none;
  }
  .minimal .help-width {
    padding: 13px 13px;
  }
```
Then let your consumer set the mode. 
```
<ia-zendesk-help-widget buttonMode="minimal"></ia-zendesk-help-widget>
```

Since it can change based on the window size, you can do a `window.onresize()` to change the property when it reaches the threshold.

Reply:
I think that will increase execution time to traverse things with window.onresize event and all. we generally can use media queries as well in CSS components so I am skipping this.

Reply:
Even though you _can_ use the media query inside a component, you are ignoring component layout encapsulation by doing so. The component, particularly one that isn't intended to take up the whole width, should only be concerned with its own layout, given its container size (or mode as the case may be), not with the entire page size. There are APIs like the [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) that help with this.